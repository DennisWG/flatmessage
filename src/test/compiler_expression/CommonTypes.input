module Playground.Net.CommonTypes;

data Vector2d
{
    float x;
    float y;
}

data Vector3d
{
    float x;
    float y;
    float z;
}