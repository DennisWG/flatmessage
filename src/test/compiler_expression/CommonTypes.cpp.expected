#include "CommonTypes.hpp"
#include <flatmessage.hpp>

using namespace Playground::Net;

BinaryEncoder& operator<<(BinaryEncoder& encoder, const Vector2d& data)
{
    encoder << data.x;
    encoder << data.y;
    return encoder;
}

BinaryDecoder& operator>>(BinaryDecoder& decoder, const Vector2d& data)
{
    decoder >> data.x;
    decoder >> data.y;
    return decoder;
}

BinaryEncoder& operator<<(BinaryEncoder& encoder, const Vector3d& data)
{
    encoder << data.x;
    encoder << data.y;
    encoder << data.z;
    return encoder;
}

BinaryDecoder& operator>>(BinaryDecoder& decoder, const Vector3d& data)
{
    decoder >> data.x;
    decoder >> data.y;
    decoder >> data.z;
    return decoder;
}
